util/testutil/workers/util.go

```diff
@@ -79,6 +79,10 @@ func runBuildkitd(
 		"BUILDKIT_DEBUG_EXEC_OUTPUT=1",
 		"BUILDKIT_DEBUG_PANIC_ON_ERROR=1",
 		"TMPDIR="+filepath.Join(tmpdir, "tmp"))
+	if v := os.Getenv("GO_TEST_COVERPROFILE"); v != "" {
```

Comment:
It seems that because `go test` sets its own `GOCOVERDIR` internally to a temp directory, a different variable needs to be used to pass a path to subprocesses.